scipy/linalg/tests/test_special_matrices.py

```diff
@@ -640,3 +640,26 @@ def test_fiedler_companion():
     fc = fiedler_companion([1., -16., 86., -176., 105.])
     assert_array_almost_equal(eigvals(fc),
                               np.array([7., 5., 3., 1.]))
+
+def test_convmtx():
+    'test convmtx matrix multiply against numpy.convolve for various parameters'
+
+    def gaussian(n, cpx):
+        'make a complex or real gaussian vector of length n'
+        if cpx:
+            return np.random.normal(size=(n, 2)).astype(np.float64).view(np.complex128).ravel()
+        else:
+            return np.random.normal(size=(n,))
+
+    for cpx, na, nv, mode in \
+            itertools.product((False, True), range(2, 7), range(2, 7), (None, 'full', 'valid', 'same')):
```

Comment:
This may be done more cleanly with a parametrized test, although it is true that we use this paradigm for testing elsewhere in a few cases.

Reply:
In order to get the same outer product coverage it now has, would one nest the `pytest.mark.parameterize` decorators?

Can you provide a reference or example?

Reply:
Indeed nesting is the typical way. 

```
@pytest.mark.parameterize('cpx', (False, True))
@pytest.mark.parameterize('na', range(2, 7))
@pytest.mark.parameterize('nv', range(2, 7))
@pytest.mark.parameterize('mode', (None, 'full', 'valid', 'same'))
def test_convmtx(cpx, na, nv, mode):
    ....
    Test contents
    .....
```

However note that this product introduces 200 tests. Hence if it takes too long we might need to pick 10, 100, 1000 as sample sizes and reduce the number of tests since they essentially test the same thing with minute differences. Testing time is unfortunately a constraint for us.

Reply:
In light of the increased number of tests, perhaps such refactoring could be postponed until there is a need for more granular tests?